konnectivity-socks5-proxy/main.go

```diff
@@ -265,7 +269,7 @@ type proxyResolver struct {
 
 func (d proxyResolver) Resolve(ctx context.Context, name string) (context.Context, net.IP, error) {
 	// Preserve the host so we can recognize it
-	if isCloudAPI(name) {
+	if isCloudAPI(name) || d.disableResolver {
```

Comment:
This won't perform dns resolution but dialling would sill go through the guest cluster and so preserving the issue described in the jira. What am I missing?

To overcome the issue you could use resolve-from-management-cluster-dns causing dialing to happen through the management cluster but then that would defeat the purpuse of the healthcheck.
Has the CNO change the healthcheck approach to e.g. run a pod on the guest cluster or similar?

Reply:
Also is there a way we can include healthchecking as part of our TestCreateClusterProxy validations?

Reply:
> This won't perform dns resolution but dialling would sill go through the guest cluster and so preserving the issue described in the jira. What am I missing?

That is intentional. The readiness check has to be performed from the guest cluster but we want to dial to the exact cluster-wide-proxy address.
I know the issue is confusing so lets consider what was happening before:
1. CNO dials to the cluster-wide-proxy fqdn using the local socks5 proxy.
2. socks5 proxy resolves the fqdn to an IP address and sends it to the guest cluster through konnectivity
3. konnectivity agent in the guest cluster has the cluster-wide-proxy fqdn in no_proxy but the cluster-wide-proxy ip is not there. It tries to dial to the cluster-wide-proxy IP through the configured proxy causing a reject from the proxy server.

With this change the behavior changes to:
1. CNO dials to the cluster-wide-proxy fqdn using the local socks5 proxy.
2. socks5 proxy doesn't resolve the fqdn and sends it to the guest cluster through konnectivity unchanged. 
3. konnectivity agent in the guest cluster has the cluster-wide-proxy fqdn in no_proxy so when it dials to it it does it directly.